internal/jobs/linkedin.go

```diff
@@ -0,0 +1,286 @@
+		return types.JobResult{Error: "invalid search type: " + jobArgs.QueryType}, fmt.Errorf("invalid search type: %s", jobArgs.QueryType)
+	}
+}
+
+func (ls *LinkedInScraper) searchProfiles(j types.Job, client *linkedinscraper.Client, args *args.LinkedInArguments) (types.JobResult, error) {
+	// Validate query is not empty
+	if args.Query == "" {
+		return types.JobResult{Error: "query is required"}, fmt.Errorf("query is required")
+	}
+
+	searchArgs := linkedinscraper.ProfileSearchArgs{
+		Keywords:       args.Query,
+		NetworkFilters: args.NetworkFilters,
+		Start:          args.Start,
+		Count:          args.MaxResults,
+	}
+
+	// Set defaults if not provided
+	if searchArgs.Count == 0 {
+		searchArgs.Count = 10
+	}
+	if len(searchArgs.NetworkFilters) == 0 {
+		searchArgs.NetworkFilters = []string{"F", "S", "O"} // All networks
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), j.Timeout)
+	defer cancel()
+
+	profiles, err := client.SearchProfiles(ctx, searchArgs)
+	if err != nil {
+		// Check for specific error types
+		if strings.Contains(err.Error(), "unauthorized") || strings.Contains(err.Error(), "401") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInAuthErrors, 1)
+		} else if strings.Contains(err.Error(), "rate limit") || strings.Contains(err.Error(), "429") {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInRateErrors, 1)
+		} else {
+			ls.statsCollector.Add(j.WorkerID, stats.LinkedInErrors, 1)
+		}
+		return types.JobResult{Error: fmt.Sprintf("failed to search profiles: %v", err)}, err
+	}
+
+	// Convert to our result type
+	var results []teetypes.LinkedInProfileResult
+	for _, profile := range profiles {
+		result := teetypes.LinkedInProfileResult{
+			PublicIdentifier: profile.PublicIdentifier,
+			URN:              profile.URN,
+			FullName:         profile.FullName,
+			Headline:         profile.Headline,
+			Location:         profile.Location,
+			ProfileURL:       profile.ProfileURL,
+			// Degree field will be empty for now since BadgeText is not available
+			Degree: "",
+		}
+		results = append(results, result)
+	}
+
+	ls.statsCollector.Add(j.WorkerID, stats.LinkedInProfiles, uint(len(results)))
+
+	data, err := json.Marshal(results)
+	if err != nil {
+		return types.JobResult{Error: "failed to marshal results"}, err
+	}
+
+	return types.JobResult{Data: data}, nil
+}
+
+func (ls *LinkedInScraper) getProfile(j types.Job, client *linkedinscraper.Client, args *args.LinkedInArguments) (types.JobResult, error) {
+	// Validate public identifier is not empty
+	if args.PublicIdentifier == "" {
+		return types.JobResult{Error: "public_identifier is required"}, fmt.Errorf("public_identifier is required")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), j.Timeout)
+	defer cancel()
+
+	profile, err := client.GetProfile(ctx, args.PublicIdentifier)
+	if err != nil {
+		// Check for specific error types
+		if strings.Contains(err.Error(), "unauthorized") || strings.Contains(err.Error(), "401") {
```

Comment:
Matching against `"401"` seems a bit fragile, but I'm not sure that we can do better. Should we try and fix it now, or add a TODO?